File: sliding-window/minimum-size-subarray-sum.js
// find the minimum length of a contiguous subarray whose sum is greater than or equal to the target
export function minSubArrayLen(target, nums) {
	let longest = Infinity;
	let sum = 0;
	let start = 0;

	for (let end = 0; end < nums.length; end++) {
		sum += nums[end];

		while (sum >= target) {
			longest = Math.min(longest, end - start + 1);

			sum -= nums[start];
			start++;
		}
	}

	return longest != Infinity ? longest : 0;
}

File: sliding-window/minimum-size-subarray-sum.py
# find the minimum length of a contiguous subarray whose sum is greater than or equal to the target
def min_sub_array_len(target, nums):
    start = 0
    curr_sum = 0
    min_length = float("inf")

    for end, val in enumerate(nums):
        curr_sum += val

        while curr_sum >= target:
            min_length = min(min_length, end - start + 1)

            curr_sum -= nums[start]
            start += 1

    return min_length if min_length != float("inf") else 0
